src/execution/window_executor.cpp

```diff
@@ -731,7 +892,15 @@ WindowAggregateExecutor::WindowAggregateExecutor(BoundWindowExpression &wexpr, C
 	} else if (wexpr.aggregate) {
 		// build a segment tree for frame-adhering aggregates
 		// see http://www.vldb.org/pvldb/vol8/p1058-leis.pdf
-		aggregator = make_uniq<WindowSegmentTree>(AggregateObject(wexpr), wexpr.return_type, count, mode);
+		if (ExclusionIgnored(wexpr)) {
```

Comment:
Maybe the exclusion setting should just get passed down into the accelerator? Then each accelerator can implement it as it sees fit (including ones we haven't created yet, like merge sort trees)?